src/core/dbus-execute.c

```diff
@@ -3584,6 +3619,83 @@ int bus_exec_context_set_transient_property(
                 extension_images = mount_image_free_many(extension_images, &n_extension_images);
 
                 return 1;
+        } else if (STR_IN_SET(name, "StateDirectorySymlink", "RuntimeDirectorySymlink")) {
+                char *source, *destination;
+                ExecDirectory *directory;
+                uint64_t symlink_flags; /* No flags for now, reserved for future uses. */
+                ExecDirectoryType i;
+                bool empty = true;
+
+                assert_se((i = exec_directory_type_symlink_from_string(name)) >= 0);
+                directory = c->directories + i;
+
+                r = sd_bus_message_enter_container(message, 'a', "(sst)");
+                if (r < 0)
+                        return r;
+
+                while ((r = sd_bus_message_read(message, "(sst)", &source, &destination, &symlink_flags)) > 0) {
+
+                        if (!utf8_is_valid(source))
```

Comment:
you can drop this check. Dbus strings are always utf8-safe. the broker (and sd-bus again) enforce that.